File: 0875-koko-eating-bananas/0875-koko-eating-bananas.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        left=1
        right=max(piles)
        result=right
        while left<=right:
            k=(left+right)//2
            hours=0
            for pile in piles:
                hours+=math.ceil(pile/k)
            if hours<=h:
                result=min(result,k)
                right=k-1
            else:
                left=k+1
        return result
        

File: 0875-koko-eating-bananas/0875-koko-eating-bananas.java
class Solution {
    public int minEatingSpeed(int[] piles, int h) {
        int left=1;
        int maxElement=0;
        for(int i=0;i<piles.length;i++){
            maxElement=Math.max(maxElement,piles[i]);
        }
        int right=maxElement;
        int minimumInteger=right;
        while(left<=right){
            int k=(left+right)/2;
            int hours=0;
            for(int pile:piles){
                hours+=Math.ceil((double)pile/k);
            }
            if(hours<=h){
                minimumInteger=Math.min(minimumInteger,k);
                right=k-1;
            }
            else{
                left=k+1;
            }
        }
        return minimumInteger;
    }
}